cmd/liqo-controller-manager/main.go

```diff
@@ -147,10 +168,66 @@ func main() {
 		os.Exit(1)
 	}
 
+	// Create an accessory manager restricted to the given namespace only, to avoid introducing
+	// performance overhead and requiring excessively wide permissions when not necessary.
+	auxmgr, err := ctrl.NewManager(config, ctrl.Options{
+		MapperProvider:     mapperUtils.LiqoMapperProvider(scheme),
+		Scheme:             scheme,
+		Namespace:          *liqoNamespace,
+		MetricsBindAddress: "0", // Disable the metrics of the auxiliary manager to prevent conflicts.
+	})
+	if err != nil {
+		klog.Fatalf("Unable to create auxiliary (namespaced) manager: %w", err)
+	}
+
+	namespacedClient := client.NewNamespacedClient(auxmgr.GetClient(), *liqoNamespace)
```

Comment:
Could be removed and replaced by a single manager with per-resource namespace constraints (if not already given cluster-wide permissions for that resource).